providers/class-two-factor-email.php

```diff
@@ -187,7 +187,7 @@ public function validate_token( $user_id, $token ) {
 		$hashed_token = $this->get_user_token( $user_id );
 
 		// Bail if token is empty or it doesn't match.
-		if ( empty( $hashed_token ) || ( wp_hash( $token ) !== $hashed_token ) ) {
+		if ( empty( $hashed_token ) || ( !hash_equals( wp_hash( $token ), $hashed_token) ) ) {
```

Comment:
Do we still need the wrapping parentheses now that is a function call?

```suggestion
		if ( empty( $hashed_token ) || ! hash_equals( wp_hash( $token ), $hashed_token ) ) {
```

Reply:
No need Of Wrapping parentheses, I have added them just to make the function and empty function look like separate ones. Will remove it to avoid confusion.